src/ASM/DomainDecomposition.C

```diff
@@ -1232,7 +1232,85 @@ bool DomainDecomposition::setup (const ProcessAdm& adm, const SIMbase& sim)
 
   if (ok < adm.getNoProcs())
     return false;
+#endif
+
+  return true;
+}
+
+
+namespace {
+
+//! \brief Wrapper class used to perform partitioning of a matrix without a SAM.
+class SAMCB
+{
+public:
+  //! \brief Constructor.
+  //! \param meqn Matrix of element equations
+  explicit SAMCB(const IntMat& meqn)
+    : m_meqn(meqn)
+  {
+      m_neq = std::accumulate(meqn.begin(), meqn.end(), 0,
+                              [](const int acc, const IntVec& meen)
+                              {
+                                const int max = *std::max_element(meen.begin(), meen.end());
+                                return std::max(acc, max);
+                              }) + 1;
+  }
+
+  //! \brief Returns the maximum number of equations.
+  size_t getNoEquations() const { return m_neq; }
```

Comment:
Unless this method is invoked many times (inside a loop of significance), one could consider letting `getNoEquation()` calculate neq instead of the constructor, and then remove the `m_neq` member.

Reply:
only called once, so done.